src/tests/ftest/util/general_utils.py

```diff
@@ -663,16 +669,34 @@ def stop_processes(hosts, pattern, verbose=True, timeout=60, added_filter=None):
     """
     result = {}
     log = getLogger()
-    log.info("Killing any processes on %s that match: %s", hosts, pattern)
+    if dump_ult_stacks is True:
+        log.info("First dumping ULT stacks, then Killing any processes on %s "
+                 "that match: %s", hosts, pattern)
+    else:
+        log.info("Killing any processes on %s that match: %s", hosts, pattern)
 
     if added_filter:
-        ps_cmd = "/usr/bin/ps x | grep -E {} | grep -vE {}".format(
+        ps_cmd = "/usr/bin/ps xa | grep -E {} | grep -vE {}".format(
             pattern, added_filter)
     else:
         ps_cmd = "/usr/bin/pgrep --list-full {}".format(pattern)
 
     if hosts is not None:
-        commands = [
+        if dump_ult_stacks is True:
+            commands_part1 = [
+                "rc=0",
+                "if " + ps_cmd,
+                "then rc=1",
+                "sudo pkill --signal USR2 {}".format(pattern),
```

Comment:
If we add the `DaosServerManager.kill_engines()` method as described above, maybe we can move this code out of the generic `stop_processes()` method and into the more server/engine-specific `kill_engines()` method.  This would allow us to execute the `sudo pkill --signal USR2 daos_engine` first,   then use some criteria - ideally besides sleep - to verify its done, and finally call the original `stop_processes()` method to just kill the `daos_engine` processes.